File: android/app/src/main/kotlin/com/example/siap/security/AppSealingThreatRegistrar.kt
package com.example.siap.security

import io.flutter.embedding.engine.FlutterEngine
import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.MethodChannel

/**
 * Registrasi Flutter platform channel.
 * Threat callback DoveRunner ditangani oleh [ThreatReceiver] (broadcast).
 */
object AppSealingThreatRegistrar {
    const val CHANNEL = "com.siap/appsealing_threats"

    fun register(flutterEngine: FlutterEngine) {
        MethodChannel(flutterEngine.dartExecutor.binaryMessenger, CHANNEL)
            .setMethodCallHandler { call, result ->
                AppSealingThreatBridge.handleMethodCall(call, result)
            }

        EventChannel(flutterEngine.dartExecutor.binaryMessenger, CHANNEL)
            .setStreamHandler(
                object : EventChannel.StreamHandler {
                    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
                        AppSealingThreatBridge.setEventSink(events)
                    }

                    override fun onCancel(arguments: Any?) {
                        AppSealingThreatBridge.setEventSink(null)
                    }
                },
            )
    }
}

File: ios/Runner/Security/AppSealingThreatRegistrar.swift
import Flutter

/// Rollback: hapus folder Security/ + pemanggilan di AppDelegate.
enum AppSealingThreatRegistrar {
    static let channelName = "com.siap/appsealing_threats"

    static func register(with registrar: FlutterPluginRegistrar) {
        let messenger = registrar.messenger()

        let methodChannel = FlutterMethodChannel(
            name: channelName,
            binaryMessenger: messenger
        )
        methodChannel.setMethodCallHandler { call, result in
            AppSealingThreatBridge.shared.handleMethodCall(call, result: result)
        }

        let eventChannel = FlutterEventChannel(
            name: channelName,
            binaryMessenger: messenger
        )
        eventChannel.setStreamHandler(AppSealingThreatStreamHandler())
    }

    static func registerAppSealingCallback() {
        // Sesuaikan dengan callback API AppSealing iOS (Xcode Guide Part IV):
        // AppSealing.shared.setHackDetectHandler { code, message, category, willKill in
        //     AppSealingThreatBridge.shared.onThreatDetected(
        //         code: code, message: message,
        //         category: category, willKillApp: willKill
        //     )
        // }
    }
}
